packages/components/src/components/notification-banner/notification-banner.tsx

```diff
@@ -0,0 +1,177 @@
+ */
+
+import {
+  Component,
+  h,
+  Host,
+  Prop,
+  Element,
+  Method,
+  State,
+} from '@stencil/core';
+import classNames from 'classnames';
+import statusNote from '../../utils/status-note';
+
+const TIMEOUT = 3000;
+
+@Component({
+  tag: 'scale-notification-banner',
+  styleUrl: 'notification-banner.css',
+  shadow: true,
+})
+export class NotificationMessage {
+  @Element() hostElement: HTMLElement;
+
+  @Prop() variant?: 'informational' | 'success' | 'warning' | 'error' =
+    'informational';
+  @Prop() dismissible?: boolean = false;
+  @Prop({ reflect: true }) opened: boolean;
+  @Prop() timeout?: boolean | number = false;
+  @Prop() href: string;
+
+  @State() hasText?: boolean;
+  @State() hasLink?: boolean;
+
+  componentDidRender() {
+    if (this.timeout) {
+      const timeout = this.timeout === true ? TIMEOUT : this.timeout;
+      setTimeout(this.close, timeout);
+    }
+    if (this.hostElement.querySelectorAll('[slot=text]').length !== 0) {
+      this.hasText = true;
+    }
+    if (this.hostElement.querySelectorAll('[slot=link]').length !== 0) {
+      this.hasLink = true;
+    }
+  }
+
+  componentDidLoad() {}
+
+  connectedCallback() {
+    statusNote({ source: this.hostElement, type: 'warn' });
+  }
+
+  @Method()
+  async open() {
+    this.opened = true;
+  }
+
+  handleIcons() {
+    if (this.variant) {
+      switch (this.variant) {
+        case 'success':
+          return (
+            <scale-notification-banner-svg
+              class="notification-banner__icon-success"
+              accessibility-title="success"
+            />
+          );
+        case 'informational':
+          return (
+            <scale-icon-alert-information
+              class="notification-banner__icon-information"
+              accessibility-title="information"
+            />
+          );
+        case 'error':
+          return (
+            <scale-icon-alert-warning
+              class="notification-banner__icon-error"
+              accessibility-title="error"
+            />
+          );
+        case 'warning':
+          return (
+            <scale-icon-alert-information
+              class="notification-banner__icon-information"
+              color="#AE461C"
+              accessibility-title="information"
+            />
+          );
+      }
+    }
+    return;
+  }
+
+  close = () => {
+    this.opened = false;
+  };
+
+  render() {
+    if (!this.opened) {
+      return null;
+    }
+
+    return (
+      <Host>
+        <div
+          part={this.getBasePartMap()}
+          class={this.getCssClassMap()}
+          tabindex="0"
+        >
+          <div part="container" class="notification-banner__container">
+            {this.handleIcons()}
+            <div part="heading" class="notification-banner__heading">
+              <slot>&emsp;</slot>
+
+              {this.dismissible && (
+                <scale-icon-action-circle-close
+                  tabindex="0"
```

Comment:
what if we wrap the icon into a button? would that improve accessibility? and it would take away the need of this tabindex.